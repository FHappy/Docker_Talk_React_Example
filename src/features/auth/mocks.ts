import { User } from "./types";

export const MOCK_USER: User = {
  id: 1,
  firstName: "Barnaby",
  lastName: "Jones",
  email: "testBoi@testtest.gov"
};
