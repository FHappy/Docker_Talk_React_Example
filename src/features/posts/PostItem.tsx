import React from "react";
import Card from "react-bootstrap/Card";

export interface PostItemProps {
  title: string;
  body: string;
}

const PostItem: React.FC<PostItemProps> = ({ title, body }) => {
  return (
    <Card>
      <Card.Body>
        <Card.Title>{title}</Card.Title>
        <Card.Body>{body}</Card.Body>
      </Card.Body>
    </Card>
  );
};

export default PostItem;
