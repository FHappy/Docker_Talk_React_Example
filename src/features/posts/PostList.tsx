import React from "react";

import PostItem from "./PostItem";
import { Post } from "./types";

export interface PostListProps {
  posts: Post[];
  title: string;
}

const PostList: React.FC<PostListProps> = ({ posts, title }) => {
  return (
    <>
      <h3 style={{ marginBottom: "1rem" }}>{title}</h3>

      <div>
        {posts.map(post => (
          <PostItem key={post.id} title={post.title} body={post.body} />
        ))}
      </div>
    </>
  );
};

export default PostList;
