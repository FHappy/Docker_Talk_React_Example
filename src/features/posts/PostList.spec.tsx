import { shallow, ShallowWrapper } from "enzyme";
import React from "react";

import { MOCK_POSTS } from "./mocks";
import PostItem from "./PostItem";
import PostList, { PostListProps } from "./PostList";

describe("Rendering the PostList component", () => {
  const mockProps: PostListProps = {
    posts: MOCK_POSTS.slice(0, 5),
    title: "Mock Posts Title"
  };

  it("Should render the PostList component correctly", () => {
    const wrapper: ShallowWrapper = shallow(<PostList {...mockProps} />);

    expect(wrapper).toMatchSnapshot();
  });

  it("Should render the PostList component with the supplied attributes", () => {
    const wrapper: ShallowWrapper = shallow(<PostList {...mockProps} />);

    const postItemHtml = wrapper
      .find("PostItem")
      .first()
      .getElement();

    expect(postItemHtml.props.title).toEqual(mockProps.posts[0].title);
    expect(postItemHtml.props.body).toEqual(mockProps.posts[0].body);
  });
});
