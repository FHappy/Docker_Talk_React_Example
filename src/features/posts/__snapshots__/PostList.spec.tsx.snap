// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rendering the PostList component Should render the PostList component correctly 1`] = `
<Fragment>
  <h3
    style={
      Object {
        "marginBottom": "1rem",
      }
    }
  >
    Mock Posts Title
  </h3>
  <div>
    <PostItem
      body="quia et suscipit
suscipit recusandae consequuntur expedita et cum
reprehenderit molestiae ut ut quas totam
nostrum rerum est autem sunt rem eveniet architecto"
      key="1"
      title="sunt aut facere repellat provident occaecati excepturi optio reprehenderit"
    />
    <PostItem
      body="est rerum tempore vitae
sequi sint nihil reprehenderit dolor beatae ea dolores neque
fugiat blanditiis voluptate porro vel nihil molestiae ut reiciendis
qui aperiam non debitis possimus qui neque nisi nulla"
      key="2"
      title="qui est esse"
    />
    <PostItem
      body="et iusto sed quo iure
voluptatem occaecati omnis eligendi aut ad
voluptatem doloribus vel accusantium quis pariatur
molestiae porro eius odio et labore et velit aut"
      key="3"
      title="ea molestias quasi exercitationem repellat qui ipsa sit aut"
    />
    <PostItem
      body="ullam et saepe reiciendis voluptatem adipisci
sit amet autem assumenda provident rerum culpa
quis hic commodi nesciunt rem tenetur doloremque ipsam iure
quis sunt voluptatem rerum illo velit"
      key="4"
      title="eum et est occaecati"
    />
    <PostItem
      body="repudiandae veniam quaerat sunt sed
alias aut fugiat sit autem sed est
voluptatem omnis possimus esse voluptatibus quis
est aut tenetur dolor neque"
      key="5"
      title="nesciunt quas odio"
    />
  </div>
</Fragment>
`;
